activerecord/test/cases/adapters/postgresql/prepared_statements_disabled_test.rb

```diff
@@ -0,0 +1,25 @@
+require "cases/helper"
+require "models/computer"
+require "models/developer"
+
+class PreparedStatementsDisabledTest < ActiveRecord::PostgreSQLTestCase
+  fixtures :developers
+
+  def setup
+    @conn = ActiveRecord::Base.establish_connection :arunit_without_prepared_statements
+  end
+
+  def teardown
+    @conn.release_connection
+    ActiveRecord::Base.establish_connection :arunit
+  end
+
+  def test_select_query_works_even_when_prepared_statements_are_disabled
+    david = developers(:david)
+
+    assert_equal false, Developer.connection.prepared_statements
+
+    developer = Developer.where(id: 1).to_a
+    assert_equal david, developer.last
```

Comment:
Can you add these two variants instead- https://github.com/rails/rails/pull/27079/commits/8c46f1f5e85640335f3e94ec776507e80fcc839f

with + without binds.